### ISSUE TITLE
Hidden nested repeating group with an empty RadioButton component still triggers validation when trying to change Task

### ISSUE THREAD

**Issue by violaguttorm** [Altinn/app-frontend-react#2474]
### Description of the bug

When using a repeating Group within a repeating Group, a hidden RadioButton-element in the inner group will still trigger generic "field is required" validation if an element is inside it, even when invisible, but only when completing the whole page/Task, not when closing the outer group.   

In the following screenshots, choosing the "Indirekte" choice makes the inner Group show up (which is `openByDefault`). Clicking it away will hide the inner Group. No validation triggers when closing the outer Group, but when I try to move on by clicking "Fullfør", the error in the second screen shot shows up. 

If this makes no sense, I hope the "Steps to Reproduce" gives a clearer picture of the issue. 

![image (5)](https://github.com/user-attachments/assets/5656fe44-945d-42d3-bcf0-6a4ce22bab60)
![image (6)](https://github.com/user-attachments/assets/cb747ade-7812-4097-8259-25b97e93f2bc)

### Steps To Reproduce

1. Create a repeating Group ("OUTER") on the last page in a Task. This one has `{"validateOnSaveRow": "All"}`.
2. Add a `Button` outside OUTER that takes you on to the next Task ("SEND IN"). 
3. Add a nested repeating Group inside OUTER ("INNER") that contains a `required` `RadioButton` ("RADIO") component. This one is `openByDefault`.
4. Add a field in OUTER with a condition for showing or hiding INNER. Observe INNER showing up
5. Trigger condition for showing INNER. 
6. Trigger condition for hiding INNER. Leave RADIO blank. 
7. Save and exit OUTER. Observe that this works. 
8. Trigger SEND IN. Observe that it complains that it says that RADIO is required.  
9. (Observe that this does not happen with an `Input`-component, even if it too is `required`.)

**Comment by olemartinorg** [comment]
Hei! Beklager sent svar herfra, @violaguttorm. Ut ifra det jeg ser i skjermbildet, handler ikke dette om at komponenten er satt opp som påkrevd, men at feltet i datamodellen (`erUtenlandskVirksomhet`). Hvis det er et påkrevd felt ifølge datamodellen, må det jo fylles ut. Hvis du har dynamikk på siden som gjør at alle måter å fylle ut det feltet på skjules bør det heller ikke markeres som påkrevd i datamodellen.

Jeg skjønner at det ikke er helt optimalt heller, så til en viss grad handler dette også om [fjerning av skjulte data](https://github.com/Altinn/app-frontend-react/issues/1709). Dersom du fjerner valget for 'indirekte' og denne repgruppa skjules (sammen med alle referanser til det påkrevde feltet inni) har vi planer om å implementere en løsning som gjør at disse dataene ryddes bort fra datamodellen før validering, men denne løsningen er ikke på plass enda.

**Comment by violaguttorm** [comment]
Ja, vi endte vel opp med å lage funksjonalitet som ryddet opp dette elementet om brukeren valgte et løp som ikke leder til denne siden. Fjerning av skjulte data er nok det vi er ute etter ja. :blush: 

Denne er vel markert som påkrevd fordi den eksisterer under en annen klasse. Dersom den klassen er satt til `null` så vil denne ikke være påkrevd, da den ikke "eksisterer". Men jeg er enig i at det gir mening at den må fylles ut dersom den eksisterer ja.

**Comment by olemartinorg** [comment]
Skjønner. Da lukker jeg denne som en slags duplikat av #1709 - det er iallefall samme konklusjon vi vil frem til her.
